src/btree/bt_read.c

```diff
@@ -223,6 +224,8 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
          ;) {
         switch (current_state = ref->state) {
         case WT_REF_DELETED:
+            if (F_ISSET(session, WT_SESSION_ROLLBACK_TO_STABLE))
+                goto read;
```

Comment:
@kommiharibabu, this change was necessary to keep RTS from skipping deleted pages during the tree walk. I first tried to set one of `WT_READ_DELETED_SKIP` or `WT_READ_DELETED_CHECK` as part of calling the walk code, but the walk code has this:
```
    /* Check whether deleted pages can be skipped. */
    if (!LF_ISSET(WT_READ_DELETED_SKIP))
        LF_SET(WT_READ_DELETED_CHECK);
```
So, if I set `WT_READ_DELETED_SKIP`, the first test in `__wt_page_in_func()` skips deleted pages, and if I don't set `WT_READ_DELETED_SKIP`, the walk code sets `WT_READ_DELETED_CHECK`, and so the second test in `__wt_page_in_func()` skips deleted pages.

The only code that sets `WT_READ_DELETED_SKIP` is `__sync_checkpoint_can_skip()`, so I was hoping you know what's going on here, why there are two flags, and if you could suggest a solution?

Reply:
@keithbostic Apologies for the delayed reply. The WT_READ_DELETED_SKIP flag is set in function `__wt_sync_file` to not to let the checkpoint read any deleted pages. With durable history, we don't read any pages into the memory as part of the checkpoint. This flag is no longer required. How about redesigning this flag and set it by RTS.

Reply:
Resolving this conversation -- the change @kommiharibabu suggested has been merged into develop and this branch updated to modify it as necessary.